src/main/java/entities/Room.java

```diff
@@ -0,0 +1,147 @@
+package entities;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Category;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.GuildChannel;
+import net.dv8tion.jda.api.entities.IPermissionHolder;
+import net.dv8tion.jda.api.entities.Invite;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.entities.VoiceChannel;
+import net.dv8tion.jda.api.requests.restaction.InviteAction;
+
+/**
+ * A logical group for a topic's text and voice channels. All mentor-mentee
+ * interactions occur within a mentoring room.
+ */
+public class Room {
+    private final Category category;
+    private final String name;
+    private final TextChannel textChannel;
+    private final VoiceChannel voiceChannel;
+
+    private static int nextRoomNumber = 1;
+
+    /**
+     * Create a new room. Existing channels for this room number (e.g. from
+     * before a bot restart) will be deleted automatically.
+     * 
+     * @param topic The Topic for this room
+     * @param number This room's number
+     * @param mentee The mentee using this room
+     */
+    public Room(Topic topic, Member mentee) {
+        this.category = topic.getCategory();
+        this.name = String.format("%s-%d", topic.getName(), nextRoomNumber);
+        nextRoomNumber++;
+
+        deleteExisting();
+
+        textChannel = category.createTextChannel(name).complete();
+        Guild guild = textChannel.getGuild();
+        ArrayList<IPermissionHolder> allowList = new ArrayList<IPermissionHolder>() {
+            // makes the linter stop complaining about not having a version UID
+            private static final long serialVersionUID = 1L;
```

Comment:
All classes should have a `serialVersionUID` for de/serialization, so that Java knows which class to deserialize data into. It can compute this automatically for most classes, but not anonymous classes like this. The Java compiler warns about this if we don't explicitly add a `serialVersionUID`, even though we aren't actually serializing this class.

I can probably replace this with a `@SuppressWarnings("serial")` to eliminate this warning.